src/test/regress/Makefile

```diff
@@ -48,6 +50,14 @@ check: check-full
 # check-full triggers all tests that ought to be run routinely
 check-full: check-multi check-multi-mx check-multi-task-tracker-extra check-worker check-follower-cluster check-failure
 
+# this can be used to print a value of variable
+# ex: make print-generated_mx_files
+print-%  : ; @echo $* = $($*)
+
+$(generated_mx_files): $(citus_abs_srcdir)/specs/%: $(citus_abs_srcdir)/specs/base_%
+	echo $@
+	cd $(citus_abs_srcdir) && cpp -undef -w -P -MMD -MP -MT $@ $< > $@
```

Comment:
Can you use the `-MMF` flag to store the makefile dependency files in a separate directory.